### PR TITLE
JENKINS-30051 - Extend description with more build results

### PR DESCRIPTION
**Pull request by dawidmalina**

### CODE REVIEW

**Comment by dawidmalina** [comment]
Changes inspired by https://github.com/Diabol/delivery-pipeline-plugin/pull/113

**Comment by dawidmalina** [comment]
@patbos What do you think about this? Early feedback will be helpful.

**Comment by patbos** [comment]
Nice! Looks like it covers https://issues.jenkins-ci.org/browse/JENKINS-22843 also.

**Comment by patbos** [comment]
Tried it out and got this:
screen shot 2015-08-16 at 20 13 37
I guess it is a missing iteration for test results.

File: src/main/java/se/diabol/jenkins/pipeline/util/PluginUtil.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     *            the plug-in to check
+     * @return <code>true</code> if the specified plug-in is installed,
+     *         <code>false</code> if not.
+     */
+    public static boolean isPluginInstalled(final String shortName) {
+        Jenkins instance = Jenkins.getInstance();
+        if (instance != null) {
+            return instance.getPlugin(shortName) != null;
+        }
+        return true;
```
Comment (patbos) [inline src/main/java/se/diabol/jenkins/pipeline/util/PluginUtil.java]: If instance is null it means that Jenkins is either initializing or on its way down and if that is the case we cant do that much so throw an IllegalStateException.

**Comment by dawidmalina** [comment]
Fixed. I have plan also add jacoco-plugin (it will cover my internal needs). What I would like to do later is to propose more generic way. Especially inside pipe.js with less code duplicates. Probably will merge all tests into one model with multiple tests result providers. If you have some idea please let me know.

**Comment by dawidmalina** [comment]
@patbos Have you ever faced such error inside injected tests:

`testCliSanity(org.jvnet.hudson.test.PluginAutomaticTestBuilder$CliSanityTest)  Time elapsed: 0.141 sec  <<< ERROR!
java.lang.NoClassDefFoundError: org/apache/maven/plugin/MojoNotFoundException`

**Comment by dawidmalina** [comment]
BTW I've added Jacoco and cobertura

**Comment by patbos** [comment]
It looks like a dependency from Jacoco. It uses the Jacoco Maven plugin to parse the coverage result file. 
For now I think junit and results from analysis-core is a good start and do Jacoco and Cobertura in another PR. 

Just an idea:
Would be nice to create a Plugin like analysis-core for coverage and if Jacoco and others could use that one for making it easier to integrate.

**Comment by dawidmalina** [comment]
Ok I will split it into 2 different PRs and maybe think about dedicated plugin:). But it won't be on my priority list.

**Comment by dawidmalina** [comment]
I moved to powermock to have ability to mock static classes.

**Comment by dawidmalina** [comment]
@patbos Tests are working on my machine but here I have this error
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.18.1:test (default-test) on project delivery-pipeline-plugin: ExecutionException: java.lang.RuntimeException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
```
Do you know what is wrong with it?

**Comment by dawidmalina** [comment]
@patbos @tommysdk change are ready to review.

**Comment by dawidmalina** [comment]
Please do a code review and merge if possible:)

**Comment by dawidmalina** [comment]
Rebase

**Comment by dawidmalina** [comment]
When this change can be merged?

**Comment by patbos** [comment]
Thanks for the contribution!

**Pull request merged**